### PR TITLE
add support for topic configurations at creation time

### PR DESCRIPTION
**Pull request by TheMeier**
Optionally support the `--config` option  at topic creation

### CODE REVIEW

File: manifests/topic.pp [hunk 1]
```diff
@@ -7,21 +7,28 @@
 # This defined type is used to manage the creation of kafka topics.
 #
 define kafka::topic(
-  $ensure             = '',
-  $zookeeper          = '',
-  $replication_factor = 1,
-  $partitions         = 1,
-  $bin_dir            = '/opt/kafka/bin',
+  String                        $ensure             = '',
```
Comment (bastelfreak) [inline manifests/topic.pp]: datatypes \o/

File: manifests/topic.pp [hunk 2]
```diff
@@ -7,21 +7,28 @@
...
+  String                        $bin_dir            = '/opt/kafka/bin',
+  Optional[Hash[String,String]] $config             = undef,
 ) {
 
   $_zookeeper          = "--zookeeper ${zookeeper}"
   $_replication_factor = "--replication-factor ${replication_factor}"
   $_partitions         = "--partitions ${partitions}"
 
+  if $config {
+    $_config = inline_template('<% @config.each do |key, value|%> --config <%= key %>=<%= value %><% end %>')
```
Comment (bastelfreak) [inline manifests/topic.pp]: it should be possible to achieve the same with the `join()` function, right?

File: spec/defines/topic_spec.rb [hunk 3]
```diff
@@ -18,12 +18,33 @@
         {
           'ensure'             => 'present',
           'zookeeper'          => 'localhost:2181',
-          'replication_factor' => '1',
```
Comment (bastelfreak) [inline spec/defines/topic_spec.rb]: hrm do we need to assume that people used quoted integers? If so, the new datatype would be a breaking change.

Comment (TheMeier) [reply spec/defines/topic_spec.rb]: How about a Variant[String, Integer]) and throw a deprecation warning on string-type?

File: manifests/topic.pp (hunk 2 above)
Comment (TheMeier) [reply manifests/topic.pp]: Hm ... yes. Will change it

**Comment by bastelfreak** [comment]
Hey @TheMeier, thanks for the PR. Please take a look at the inline comments I made. Also check the used email address in the commit, it isn't associated with your github account.

File: spec/defines/topic_spec.rb (hunk 3 above)
Comment (bastelfreak) [reply spec/defines/topic_spec.rb]: sounds good!

**Comment by TheMeier** [comment]
@bastelfreak all done :)

**Pull request merged**